### ISSUE TITLE
Dracolich template issues

### ISSUE THREAD

**Issue by dungeonguru** [Sklore/HL_DD_5e_Colab#600]
Dracolich template isn't applying the undead damage and condition immunities. Reverting out of the template doesn't change race name back to dragon (remains dracolich).

**Comment by Sklore** [comment]
Does anyone know how you apply a template? I can't even find the option.

**Comment by dungeonguru** [comment]
You apply a template on the race tab. It only shows up for NPCs - they can't be applied to heroes as far as I've found. It is almost easier to build a custom race for one-off characters or monsters right now since the addition and removal of templates is buggy.

**Comment by grandchefpanda** [comment]
It's in the eval script of the Race Template Dracolich, the only way I had found to change the name was to replace dragon by dracolich in the livename of the Base Race. Apparently it is not reevaluated if the template is removed.

**Comment by dungeonguru** [comment]
The main issue I am seeing in the template are in the tags section - the DamageImm, DamComImm and other damage resistance tags don't show up on the character sheet or special tabs after applying a template. I was trying to create a template set for the lycanthropes and used the dracolich template as a starting point. The name replacement is annoying but not sure if it would be simple to reverse out an eval script or if it's a feature that the devs are looking at even. But, the resistances and immunities not working is a bit odd.

**Comment by dungeonguru** [comment]
I found a fix for the tags, I moved them from tags into an eval script as follows if someone wants to add to the community file (I put the resistances in at First but they work at most of the early timing levels):

```
<eval phase="First" priority="10000" index="2">~set the new resistances and immunities
perform hero.assign[DamageRes.dtNecrotic]
perform hero.assign[DamageImm.dtPoison]
perform hero.assign[CondImm.pcnCharmed]
perform hero.assign[CondImm.pcnExhaust]
perform hero.assign[CondImm.pcnFright]
perform hero.assign[CondImm.pcnParaly]
perform hero.assign[CondImm.pcnPoison]</eval>
```

That just means the name keeps static even after removing the template.

**Comment by mjolnirh** [comment]
I just noticed that the race spells and spell tabs do not show up when this template is added, even though xspellcaster thing is added to the template ablilities.

**Comment by DarkHayZe** [comment]
Far as I can tell this is a LW bug, and there is nothing we can do about it. Unless I am just missing it we do not even have a community MM file.

**Comment by mjolnirh** [comment]
well I've been messing with it and it seems that the template function does not have a caster level variable that it can transfer over to the subject of the template.  I could get the spells tab to come up, but no spell slots were available.  So most likely it would involve an eval script similar to this one

perform linkage[table].setfocus
doneif (state.isfocus = 0)

doneif (tagis[Helper.Disable] <> 0)

~ Spellcasting attribute
perform focus.setlinkage[castattr,BaseAttr,"IsAttr.aINT"]

~ Spellcasting type
perform focus.assign[CasterType.SpontKnow]
perform focus.assign[CasterSrc.Arcane]
perform focus.assign[Helper.3rdCaster]
perform focus.assign[sClass.cHelpWiz]

~ Cantrip Array
focus.field[cArrKnCan].arrayvalue[2] += 2
focus.field[cArrKnCan].arrayvalue[9] += 3

~ Spells Known Array
focus.field[cArrKnSpl].arrayvalue[2] += 3
focus.field[cArrKnSpl].arrayvalue[3] += 4
focus.field[cArrKnSpl].arrayvalue[6] += 5
focus.field[cArrKnSpl].arrayvalue[7] += 6
focus.field[cArrKnSpl].arrayvalue[9] += 7
focus.field[cArrKnSpl].arrayvalue[10] += 8
focus.field[cArrKnSpl].arrayvalue[12] += 9
focus.field[cArrKnSpl].arrayvalue[13] += 10
focus.field[cArrKnSpl].arrayvalue[15] += 11
focus.field[cArrKnSpl].arrayvalue[18] += 12
focus.field[cArrKnSpl].arrayvalue[19] += 13

~ Max Spell Levels Array
focus.field[cArrKnLev].arrayvalue[2] += 1
focus.field[cArrKnLev].arrayvalue[6] += 2
focus.field[cArrKnLev].arrayvalue[12] += 3
focus.field[cArrKnLev].arrayvalue[18] += 4

but this will not work for a race

**Comment by DarkHayZe** [comment]
~ Set spell casting attribute
       perform hero.childfound[cHelpMnk].assign[SplAttr.aWIS]

I would suggest this for your assigning spell casting attribute line

**Comment by mjolnirh** [comment]
got the spell slots to show up but no tab to choose spells

var wcslvl as number

      wcslvl = herofield[tHitDice].value

	if (wcslvl >= 21) then
        wcslvl = 20
        endif

hero.childfound[SpellSlots].field[FinalSpLev].value = wcslvl

**Comment by TheEnthusiast** [comment]
Set Hero.Caster

Enviado de meu ASUS

**Comment by dungeonguru** [comment]
> Far as I can tell this is a LW bug, and there is nothing we can do about it. Unless I am just missing it we do not even have a community MM file.

I thought the template was a  community thing - t5CDraLich, not sure what file it is in but the "5C" is supposed to be community side as far as I knew. 

Is a contributor collecting all the scripting so far? Do I need to open a seperate bug for LW?

**Comment by grandchefpanda** [comment]
The MM is completely in the community pack, all monsters and yes, the Dracolish template is a community stuff too. The problem is that there is no other examples except the 3 simples templates that I did because there is no template in the SRD.

**Comment by mjolnirh** [comment]
perform hero.assign[Hero.Caster] brings up the spells tab that shows the spell slots, but not the tab that will allow you to choose the spells.

**Comment by dungeonguru** [comment]
I'm going to take another crack at this since I got used to putting spell lists and slots on monsters for Volo's. I noticed that the Dracolich template is in the PHB file and there isn't a MM community file - I'll go through my MM and see if we're missing anything else and create a new file if there is a need. I have some templates I've created to add High Elf, Wood Elf, etc. to the NPC races that would make a nice addition if I can get dracolich working right, minus the rename issue that I'll open with LW if I can confirm where exactly the bug starts.
